static/app/components/events/eventEntries.tsx

```diff
@@ -126,62 +127,60 @@ const EventEntries = memo(
         error_message: uniq(errorMessages),
         ...(platform && {platform}),
       });
-    }
+    }, [event, organization, project.platform]);
 
-    async function fetchProguardMappingFiles(query: string): Promise<Array<DebugFile>> {
-      try {
-        const proguardMappingFiles = await api.requestPromise(
-          `/projects/${orgSlug}/${projectSlug}/files/dsyms/`,
-          {
-            method: 'GET',
-            query: {
-              query,
-              file_formats: 'proguard',
-            },
-          }
-        );
-        return proguardMappingFiles;
-      } catch (error) {
-        Sentry.captureException(error);
-        // do nothing, the UI will not display extra error details
-        return [];
-      }
-    }
-
-    function isDataMinified(str: string | null) {
-      if (!str) {
-        return false;
-      }
+    const fetchProguardMappingFiles = useCallback(
+      async (query: string): Promise<Array<DebugFile>> => {
+        try {
+          const proguardMappingFiles = await api.requestPromise(
+            `/projects/${orgSlug}/${projectSlug}/files/dsyms/`,
+            {
+              method: 'GET',
+              query: {
+                query,
+                file_formats: 'proguard',
+              },
+            }
+          );
+          return proguardMappingFiles;
+        } catch (error) {
+          Sentry.captureException(error);
+          // do nothing, the UI will not display extra error details
+          return [];
+        }
+      },
+      [api, orgSlug, projectSlug]
+    );
 
-      return !![...str.matchAll(MINIFIED_DATA_JAVA_EVENT_REGEX_MATCH)].length;
-    }
+    const hasThreadOrExceptionMinifiedFrameData = useCallback(
```

Comment:
If this doesn't have any dependencies should we just pull it out of the component?

Reply:
hoisted